lib/Parse/ParseStmt.cpp

```diff
@@ -2529,3 +2548,66 @@ ParserResult<Stmt> Parser::parseStmtPoundAssert() {
   return makeParserResult<Stmt>(new (Context) PoundAssertStmt(
       SourceRange(startLoc, endLoc), conditionExprResult.get(), message));
 }
+
+//
+///// Produce a score (smaller is better) comparing a parameter name and
+///// potentially-typo'd argument name.
+/////
+///// \param identifier The name of the identifier.
+///// \param stmt The name of the argument.
+///// \param maxScore The maximum score permitted by this comparison, or
+///// 0 if there is no limit.
+/////
+///// \returns the score, if it is good enough to even consider this a match.
+///// Otherwise, an empty optional.
+/////
+static Optional<unsigned> scoreIdentifierAndStmtNameTypo(StringRef identifier,
```

Comment:
There are identical function used for matching stringRefs in CSSimplify.cpp. Can it be moved to StringExtras?

Reply:
I think you can get away with a table of keywords and a loop checking `edit_distance`.

Reply:
Agree. Fixed.

Reply:
Agree. Fixed.